client/python/datajunction/client.py

```diff
@@ -1,11 +1,19 @@
 """DataJunction client setup."""
+# pylint: disable=redefined-outer-name, import-outside-toplevel
 import enum
 import logging
 import platform
+import warnings
 from typing import Any, Dict, List, Optional, Union
 from urllib.parse import urljoin
 
-import pandas as pd
+try:
+    import pandas as pd
+except ImportError:
+    warnings.warn(
+        "Optional dependency `pandas` not found, data retrieval disabled",
```

Comment:
Should this include some directions on how to install pandas?

Reply:
That's a good idea, let me update that 👍